### PR TITLE
Use werkzeug make_server for starting & shutdown

### PR DESCRIPTION
**Pull request by T4rk1n**
Replace the start of the server by calling `make_server` from werkzeug which can then be used to `shutdown` the server properly.

### CODE REVIEW

File: jupyter_dash/jupyter_app.py [hunk 1]
```diff
@@ -302,25 +303,32 @@ def run(
...
+        logging.getLogger("werkzeug").setLevel(logging.ERROR)
+
         @retry(
             stop_max_attempt_number=15,
             wait_exponential_multiplier=100,
             wait_exponential_max=1000
         )
         def run():
             try:
-                super_run_server(**kwargs)
```
Comment (alexcjohnson) [inline jupyter_dash/jupyter_app.py:312]: these `**kwargs` were previously used to enable devtools, right? Can we just explicitly call `self.enable_dev_tools` with the appropriate subset of kwargs? And then perhaps throw an error if there are any unused kwargs left?

Comment (T4rk1n) [reply jupyter_dash/jupyter_app.py:312]: Added `enable_dev_tools` call. :ok_hand:

**Comment by alexcjohnson** [comment]
OK maybe a little unlikely, mostly this works great in my testing, but I did find a way to break it. I noticed when I ran two apps, one on the default port and one on `port=8051`, then `app._servers` looked like this:
```py
{('127.0.0.1', '8050'): <werkzeug.serving.ThreadedWSGIServer at 0x121b394e0>,
 ('127.0.0.1', 8051): <werkzeug.serving.ThreadedWSGIServer at 0x1226cdde0>}
```
So then I tried to rerun the second app, but instead of `port=8051` I used `port='8051'`, and THAT gave:
```
OSError: [Errno 48] Address already in use
```
Simple fix, just coerce port to be an integer instead of a string.

**Review by alexcjohnson — APPROVED** [review]
💃 after considering https://github.com/plotly/jupyter-dash/pull/105#issuecomment-1286225071

**Comment by aiqc** [comment]
Does this also address the dreaded https://github.com/plotly/jupyter-dash/issues/33?

**Pull request merged**